components/fxa-client/android/src/main/java/mozilla/appservices/fxaclient/FirefoxAccount.kt

```diff
@@ -384,18 +378,37 @@ class FirefoxAccount(handle: FxaHandle, persistCallback: PersistCallback?) : Aut
 
     /**
      * Ensure that the supported capabilities described earlier in `initializeDevice` are A-OK.
+     * Optionally, the caller can also pass a set of capabilities which will be validated on top of the
+     * previously registered ones. This is useful as in a migration scenario where `initializeDevice`
+     * is never be called.
+     *
      * As for now there's only the Send Tab capability, so we ensure the command is registered with the server.
      * This method should be called at least every time the sync keys change (because Send Tab relies on them).
      *
      * This performs network requests, and should not be used on the main thread.
      */
-    fun ensureCapabilities() {
+    fun ensureCapabilities(additionalCapabilities: Set<Device.Capability>?) {
```

Comment:
Calling them `additional` is a little confusing to me. Can't this just take all of the capabilities that consumer supports?  It's not clear to me from this method signature what should happen if you're _not_ upgrading. My first instinct is that this would expect to receive `null`, which makes consuming this API more onerous.

Also, maybe it's worth calling out that currently there's no downgrade path - e.g. you can't "deregister" a capability. That's fine, but perhaps deserves an explicit doc mention?